ext/standard/fsock.c

```diff
@@ -73,14 +73,27 @@ static void php_fsockopen_stream(INTERNAL_FUNCTION_PARAMETERS, int persistent)
 	}
 
 	/* prepare the timeout value for use */
+	if (timeout < 0.0 || timeout > (double) PHP_TIMEOUT_ULL_MAX / 1000000.0) {
```

Comment:
This should still be the following to catch NAN and INF
```suggestion
	if (!(timeout >= 0.0 && timeout <= (double) PHP_TIMEOUT_ULL_MAX / 1000000.0)) {
```

Sorry for the confusion, this is in fact something I missed in my original patch. So I opened a PR to fix that: https://github.com/php/php-src/pull/15061

However, I'm not sure this is the best approach here after all.
My original patch didn't use a range check because some people use -1 to mean a practically infinite timeout, and I think we should keep that behaviour here as well. WDYT?